### PR TITLE
Fixed unresolved types and code cleanup

### PR DESCRIPTION
**Pull request by midnightercz**
- Removed ##type: ignore 
- Removed commented code

### CODE REVIEW

File: pubtools/_quay/image_untagger.py [hunk 1]
```diff
@@ -106,22 +109,27 @@ def construct_tag_digest_mappings(
...
+            cast(QuayClient, self._quay_client).get_repository_tags(repository),
+        )
 
-        for tag in repo_tags["tags"]:  # type: ignore
+        for tag in repo_tags["tags"]:
             image = image_schema.format(self.host, repository, tag)
             try:
-                manifest = self._quay_client.get_manifest(image)  # type: ignore
+                manifest = cast(
+                    Manifest | Manifest, cast(QuayClient, self._quay_client).get_manifest(image)
```
Comment (lrossi-dev) [inline pubtools/_quay/image_untagger.py:121]: Should it be `Manifest | ManifestList`?

File: pubtools/_quay/signer_wrapper.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ def __init__(
             settings (dict): Settings for the signer.
         """
         self.config_file = config_file
-        self.settings = settings
+        self.settings = settings or {}
```
Comment (lrossi-dev) [inline pubtools/_quay/signer_wrapper.py:55]: Can we replace this with `settings: Dict[str, Any] = {}` in method signature?

**Review by lrossi-dev — APPROVED** [review]

File: pubtools/_quay/signer_wrapper.py [hunk 3]
```diff
@@ -239,10 +239,9 @@ def _run_store_signed(self, signed_results: Dict[str, Any]) -> None:
         Signatures are uploaded in batches.
 
         Args:
-            claim_messages ([dict]):
-                Signature claim messages constructed for the RADAS service.
-            signature_messages ([dict]):
-                Messages from RADAS containing image signatures.
+            signed_results: (Dict[str, Any]):
+                Dictionary of {"operation":..., "operation_results":..., "signing_key":...}"}
```
Comment (emilyzheng) [inline pubtools/_quay/signer_wrapper.py:243]: https://github.com/release-engineering/pubtools-sign/blob/main/src/pubtools/sign/signers/msgsigner.py#L462
The entrypoint returns signer_result, operation_results and signing_key.

File: pubtools/_quay/security_manifest_pusher.py [hunk 4]
```diff
@@ -564,4 +566,19 @@ def push_security_manifests(self) -> None:
             ]
             for future in futures.as_completed(future_results):
                 if future.exception():
-                    raise future.exception()  # type: ignore # pragma: no cover
+                    raise cast(BaseException, future.exception())  # pragma: no cover
+        #     m,
```
Comment (emilyzheng) [inline pubtools/_quay/security_manifest_pusher.py:570]: Remove these?

File: pubtools/_quay/push_docker.py [hunk 5]
```diff
@@ -648,7 +666,7 @@ def run(self) -> None:
                 )
                 # TOFIX: current_signatures can contain different types of data
```
Comment (emilyzheng) [inline pubtools/_quay/push_docker.py:667]: Is it fixed?

File: pubtools/_quay/image_untagger.py (hunk 1 above)
Comment (midnightercz) [reply pubtools/_quay/image_untagger.py:121]: yeah, that's correct

File: pubtools/_quay/signer_wrapper.py (hunk 2 above)
Comment (midnightercz) [reply pubtools/_quay/signer_wrapper.py:55]: yeah, that would be ok from typing perspective but having empty dict or list as default in kwargs is considered as dangerous according to some PEP

File: pubtools/_quay/signer_wrapper.py (hunk 3 above)
Comment (midnightercz) [reply pubtools/_quay/signer_wrapper.py:243]: yeah, let me fix that

File: pubtools/_quay/push_docker.py (hunk 5 above)
Comment (midnightercz) [reply pubtools/_quay/push_docker.py:667]: yeah, that should be fixed

**Review by emilyzheng — APPROVED** [review]

**Pull request merged**